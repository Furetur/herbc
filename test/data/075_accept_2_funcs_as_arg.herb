fn filteredRange(filter: (int) -> bool, map: (int) -> int) {
    var i = 0;
    while i < 100 {
        if filter(i) {
            print(map(i));
            print(",");
        }
        i = i + 1;
    }
}

fn is_prime(x: int) -> bool {
    var i = 2;

    while i * i <= x {
        if x % i == 0 {
            return false;
        }
        i = i + 1;
    }
    return true;
}

fn square(x: int) -> int {
    return x * x;
}

entrypoint {
    filteredRange(is_prime, square);
}
