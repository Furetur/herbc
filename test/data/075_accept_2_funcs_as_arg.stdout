0,1,4,9,25,49,121,169,289,361,529,841,961,1369,1681,1849,2209,2809,3481,3721,4489,5041,5329,6241,6889,7921,9409,